prawns/evm-contracts/src/evm-lib/getContractType.ts

```diff
@@ -0,0 +1,32 @@
+// todo look into ousing this: https://www.npmjs.com/package/evm
+
+import * as ethers from 'ethers';
+import CONTRACT_METHODS, { ContractType } from './contractMethods';
+
+export function getContractType(methodsInContract: string[]): ContractType {
+  const found = Object.entries(CONTRACT_METHODS).find(([type, methods]) => {
+    return methods.every((method) => methodsInContract.includes(method));
```

Comment:
I think this might run into issues if we have contract type which are extended from existing specifications.

Like if contract type a1 contains method 'foo' and contract type a2 contains methods 'foo' and 'bar' and you pass in a contract that has methods 'foo' and 'bar' then depending on the order of the CONTRACT_METHODS you will either get back 'a1' or 'a2'.

Presumably in that scenario you actually want 'a1' and 'a2' as the contract is compliant with both standards?

Reply:
If we only want it to return one contract then I guess we could just make it a rule that in the scenario above 'a2' is always first in CONTRACT_METHODS

Reply:
As discussed some kind of one to many tagging system and a new schema needs looking into